README.md

```diff
@@ -156,3 +156,46 @@ foobar_adder <- function(foo, bar, baz = deprecated()) {
   foo + bar
 }
 ```
+
+
+### Workflow
+
+#### Where do these deprecation warnings come from?
+
+Call `lifecycle::last_warnings()` to see backtraces for all the deprecation warnings that were issued during the last top-level command.
+
+
+#### Bumping deprecation stage
+
+Some manual search and replace is needed to bump the status of deprecated features. We recommend starting with defunct features and work your way up:
+
+1. Search for `deprecate_stop()` and remove the feature from the package. The feature is now archived.
+
+1. Search for `deprecate_warn()` and replace with `deprecate_stop()`.
+
+1. Search for `deprecate_soft()` and replace with `deprecate_warn()`.
+
+1. Call `deprecate_soft()` from newly deprecated functions.
+
+Don't forget to update the badges in the documentation topics.
+
+
+### Test for deprecated features
+
+You can test that a deprecated feature still works by disabling the warnings with `scoped_lifecycle_silence()`:
+
+```{r}
+test_that("`baz` argument of `foobar_adder()` still works", {
+  scoped_lifecycle_silence()
+  foo(1, baz = 2)
+})
+```
+
+You can test that a feature is deprecated by forcing the warnings or forcing errors:
```

Comment:
I think this should get a separate header, e.g. "Find deprecated functions used by your package".

Reply:
This is about testing a function/argument is deprecated though. Or maybe we don't recommend testing for this?

Reply:
Oooh sorry, misread it. Wouldn't you just do

```R
expect_warning(foo(1, baz = 2), class = "lifecycle_deprecated")
```

? 

(i.e. I don't think that you should need to force an error, and I think the class name might be wrong)

Reply:
Oh it's `defunctError` to match base R? We should probably still subclass to give us flexibility going forward.

Reply:
> Wouldn't you just do
> 
> expect_warning(foo(1, baz = 2), class = "lifecycle_deprecated")

yes this should work most of the time unless it's the second time you capture it, then `scoped_lifecycle_warnings()` might be needed.

> Oh it's defunctError to match base R? We should probably still subclass to give us flexibility going forward.

About `defunctError`, I was thinking we should contribute a constructor to base R. It would explictly allow subclassing, and would also be more flexible regarding the feature specification (either a function, or a function argument, both for `old` and `new` fields).

In the meantime I'll add a `lifecycle_error_deprecated` subclass.